SUSYBSMAnalysis/HSCP/src/BetaCalculatorECAL.cc

```diff
@@ -321,7 +323,7 @@ void BetaCalculatorECAL::addStepToXtal(std::map<int,GlobalPoint>& trackExitPosit
     const CaloSubdetectorGeometry* theSubdetGeometry)
 {
 
-  const CaloCellGeometry *cell_p = theSubdetGeometry->getGeometry(aDetId);
+  const CaloCellGeometry *cell_p = (theSubdetGeometry->getGeometry(aDetId)).get();
```

Comment:
Remove `get` and hold onto the `std::shared_ptr` to avoid the temporary going away.